### DISCUSSION TITLE
How to Customize the Archive Page for a Blog Plugin

### DISCUSSION THREAD

**Discussion by shenweiyan** [squidfunk/mkdocs-material#6324] [Q&A] [answered]
Hi mkdocs-material,
I want to customize the archive page of the mkdocs-material blog plugin, as shown in the screenshot below. Could anyone please advise on the steps to achieve this? Thank you.

![blog-archive](https://github.com/squidfunk/mkdocs-material/assets/26101369/68b23e34-bbc1-4105-96ac-af3ec8f89fd2)

**Comment by alexvoss** [comment]
It should be possible to achieve this with just CSS by changing `display` to `inline` or `none` for a bunch of things. Try this out using the web developed tools in your browser.

I managed to change the appearance on one of my pages. See the two different stylings for the two posts visible in the screenshot below. Changing the format of the date is a different question. You could need to tell the blog plugin to produce different text, which, I guess, can be done using the [post_url_date_format](https://squidfunk.github.io/mkdocs-material/plugins/blog#config.post_url_date_format) setting?

<img width="856" alt="a screenshot showing two different styles for blog posts listed on an index page" src="https://github.com/squidfunk/mkdocs-material/assets/4134224/1c20442a-bfc7-4de3-af17-e15c01f6bc53">

**Reply by squidfunk** [reply, marked as answer]
This, or you can even set another template for archive pages and provide your own template. The blog plugin uses the same mechanics as the rest of Material for MkDocs: simple Jinja2 templating. I haven't tested it, but you should be able to do something like this using MkDocs' [hooks](https://www.mkdocs.org/user-guide/configuration/#hooks):

``` py
from material.plugins.blog.structure import Archive
def on_page_markdown(markdown, *, page, config, files):
    if isinstance(page, Archive):
        page.meta["template"] = "my-custom-template.html"
```

**Reply by shenweiyan** [reply]
Thanks, this is a great method, but it seems doesn't work for pages beyond the second page if there is pagination.

**Reply by squidfunk** [reply]
You're right! If you want this fixed, you can create a [bug report](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/) with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) – we're happy to look into it. I think we need to change this logic to take the type of the view into account:

https://github.com/squidfunk/mkdocs-material/blob/9ff75d11838cc01615697884c0eb9eb55f4652ad/src/plugins/blog/plugin.py#L622-L623

**Reply by shenweiyan** [reply]
Thank you very much for your enthusiastic assistance. This is a minimal reproduction.

Repo: <https://github.com/shenweiyan/mkdocs-blog>
Link: <https://shenweiyan.github.io/mkdocs-blog/blog/archive/2023/>

**Reply by squidfunk** [reply]
Thanks. However, we need you to go through our [bug reporting process](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/). Please carefully read the guide and provide all the necessary information. This will enable us to help you as quickly as possible. Thank you!

**Reply by shenweiyan** [reply]
Thank you very much. I have created an issue https://github.com/squidfunk/mkdocs-material/issues/6336.
